### PR TITLE
Assignment#1

### PR DESCRIPTION
**Pull request by Harinesh5**

### CODE REVIEW

File: Assignments/Harinesh/Assignment-1  (Harinesh MA A).txt [hunk 1]
```diff
@@ -0,0 +1,139 @@
+                                                                                    Assignment-1  (Harinesh MA A)
+
+1)total no of employees
+Ans: select count(ename) "total employees" 
+    from emp;
+
```
Comment (raviproengage) [inline Assignments/Harinesh/Assignment-1  (Harinesh MA A).txt:6]: OK

select count(*) from emp;

File: Assignments/Harinesh/Assignment-1  (Harinesh MA A).txt [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+2)total no of departments
+Ans: select count(deptno) "total department" 
+    from dept;
+
```
Comment (raviproengage) [inline Assignments/Harinesh/Assignment-1  (Harinesh MA A).txt:10]: OK.
Select count(*) from dept;

File: Assignments/Harinesh/Assignment-1  (Harinesh MA A).txt [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+3)all of employees
+Ans: select * 
+    from emp;
+
```
Comment (raviproengage) [inline Assignments/Harinesh/Assignment-1  (Harinesh MA A).txt:14]: ok

File: Assignments/Harinesh/Assignment-1  (Harinesh MA A).txt [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+4)all of departments
+Ans: select * 
+    from dept;
+
```
Comment (raviproengage) [inline Assignments/Harinesh/Assignment-1  (Harinesh MA A).txt:18]: ok

File: Assignments/Harinesh/Assignment-1  (Harinesh MA A).txt [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+5)total salary paid for all employees
+Ans: select sum(sal) "total salary" 
+    from emp;
+
```
Comment (raviproengage) [inline Assignments/Harinesh/Assignment-1  (Harinesh MA A).txt:22]: ok

File: Assignments/Harinesh/Assignment-1  (Harinesh MA A).txt [hunk 6]
```diff
@@ -0,0 +1,139 @@
...
+6)total commission paid for all employees
+Ans: select sum(comm) "Total commission" 
+    from emp;
+
```
Comment (raviproengage) [inline Assignments/Harinesh/Assignment-1  (Harinesh MA A).txt:26]: ok

File: Assignments/Harinesh/Assignment-1  (Harinesh MA A).txt [hunk 7]
```diff
@@ -0,0 +1,139 @@
...
+7)Which job titles of the employees get commission paid,
+Ans: select job,comm 
+    from emp 
+    where comm>0;
+
```
Comment (raviproengage) [inline Assignments/Harinesh/Assignment-1  (Harinesh MA A).txt:31]: not ok.

select distinct job from emp where comm>0;

File: Assignments/Harinesh/Assignment-1  (Harinesh MA A).txt [hunk 8]
```diff
@@ -0,0 +1,139 @@
...
+8)System date
+Ans: select current_timestamp "System date" 
+    from dual;
+
```
Comment (raviproengage) [inline Assignments/Harinesh/Assignment-1  (Harinesh MA A).txt:35]: not ok. only date asked.

select sysdate from dual;

File: Assignments/Harinesh/Assignment-1  (Harinesh MA A).txt (hunk 7 above)
Comment (raviproengage) [reply Assignments/Harinesh/Assignment-1  (Harinesh MA A).txt:31]: only job title asked

File: Assignments/Harinesh/Assignment-1  (Harinesh MA A).txt [hunk 9]
```diff
@@ -0,0 +1,139 @@
...
+9)Average salary paid to all employees.
+Ans: select avg(sal) as "Average salery" 
+    from emp;
+
```
Comment (raviproengage) [inline Assignments/Harinesh/Assignment-1  (Harinesh MA A).txt:39]: ok

File: Assignments/Harinesh/Assignment-1  (Harinesh MA A).txt [hunk 10]
```diff
@@ -0,0 +1,139 @@
...
+10)How many employees are there in each department?
+Ans: select deptno,count(*) "no_of_emp" 
+    from emp 
+    group by deptno;
+
```
Comment (raviproengage) [inline Assignments/Harinesh/Assignment-1  (Harinesh MA A).txt:44]: ok

File: Assignments/Harinesh/Assignment-1  (Harinesh MA A).txt [hunk 11]
```diff
@@ -0,0 +1,139 @@
...
+11)Total salary of the employees in each department
+Ans: select deptno,sum(sal) "no of emp" 
+    from emp 
+    group by deptno;
+
```
Comment (raviproengage) [inline Assignments/Harinesh/Assignment-1  (Harinesh MA A).txt:49]: ok

File: Assignments/Harinesh/Assignment-1  (Harinesh MA A).txt [hunk 12]
```diff
@@ -0,0 +1,139 @@
...
+12)How many employees are there under each job title?
+Ans: select job,count(*) "no of emp" 
+    from emp 
+    group by job;
+
```
Comment (raviproengage) [inline Assignments/Harinesh/Assignment-1  (Harinesh MA A).txt:54]: ok

**Comment by raviproengage** [comment]
ok

File: Assignments/Harinesh/Assignment-1  (Harinesh MA A).txt [hunk 13]
```diff
@@ -0,0 +1,139 @@
...
+13)Average salary paid for each job title.
+Ans: select job,avg(sal) 
+    from emp 
+    group by job;
+
```
Comment (raviproengage) [inline Assignments/Harinesh/Assignment-1  (Harinesh MA A).txt:59]: ok

File: Assignments/Harinesh/Assignment-1  (Harinesh MA A).txt [hunk 14]
```diff
@@ -0,0 +1,139 @@
...
+14)hire day, month, and year for each employee
+Ans: select ename,hiredate,to_char(hiredate,'DD') as hire_date,to_char
+    (hiredate,'MM') as hire_month,to_char(hiredate,'YYYY') as hire_year from emp;
+
```
Comment (raviproengage) [inline Assignments/Harinesh/Assignment-1  (Harinesh MA A).txt:63]: ok

File: Assignments/Harinesh/Assignment-1  (Harinesh MA A).txt [hunk 15]
```diff
@@ -0,0 +1,139 @@
...
+15)Sort the employee department wise.
+Ans: select * 
+    from emp 
+    order by deptno;
+
```
Comment (raviproengage) [inline Assignments/Harinesh/Assignment-1  (Harinesh MA A).txt:68]: ok

File: Assignments/Harinesh/Assignment-1  (Harinesh MA A).txt [hunk 16]
```diff
@@ -0,0 +1,139 @@
...
+16)Sort the employee based on their job titles.
+Ans: select * 
+    from emp 
+    order by job;
+
```
Comment (raviproengage) [inline Assignments/Harinesh/Assignment-1  (Harinesh MA A).txt:73]: ok

File: Assignments/Harinesh/Assignment-1  (Harinesh MA A).txt [hunk 17]
```diff
@@ -0,0 +1,139 @@
...
+17)Sort the employee based on descending order of their salaries.
+Ans: select * 
+    from emp 
+    order by sal desc;
+
```
Comment (raviproengage) [inline Assignments/Harinesh/Assignment-1  (Harinesh MA A).txt:78]: ok

File: Assignments/Harinesh/Assignment-1  (Harinesh MA A).txt [hunk 18]
```diff
@@ -0,0 +1,139 @@
...
+18)Sort the employee ascending order of their department and descending order of their salary.
+Ans: select * 
+    from emp 
+    order by deptno,sal desc;
+
```
Comment (raviproengage) [inline Assignments/Harinesh/Assignment-1  (Harinesh MA A).txt:83]: ok

File: Assignments/Harinesh/Assignment-1  (Harinesh MA A).txt [hunk 19]
```diff
@@ -0,0 +1,139 @@
...
+19)How many employees have their name with 6 characters.
+Ans: select count(ename) "name with 6 characters" from emp where Ename like '______%';
+
```
Comment (raviproengage) [inline Assignments/Harinesh/Assignment-1  (Harinesh MA A).txt:86]: not ok.

select count(*) from emp where length(ename)=6;

File: Assignments/Harinesh/Assignment-1  (Harinesh MA A).txt [hunk 20]
```diff
@@ -0,0 +1,139 @@
...
+20)Maximum and minimum salary paid.
+Ans: select max(sal),min(sal) 
+    from emp;
+
```
Comment (raviproengage) [inline Assignments/Harinesh/Assignment-1  (Harinesh MA A).txt:90]: ok

File: Assignments/Harinesh/Assignment-1  (Harinesh MA A).txt [hunk 21]
```diff
@@ -0,0 +1,139 @@
...
+21)Maximum, minimum, average and sum of salary paid under each department
+Ans: select deptno,max(sal) "max",min(sal) "min",avg(sal) "avg",sum(sal) "sum" 
+    from emp 
+    group by deptno;
+
```
Comment (raviproengage) [inline Assignments/Harinesh/Assignment-1  (Harinesh MA A).txt:95]: ok

File: Assignments/Harinesh/Assignment-1  (Harinesh MA A).txt [hunk 22]
```diff
@@ -0,0 +1,139 @@
...
+22)Sort the employee based on their hire date.
+Ans:select * 
+    from emp 
+order by hiredate;
+
```
Comment (raviproengage) [inline Assignments/Harinesh/Assignment-1  (Harinesh MA A).txt:100]: ok

File: Assignments/Harinesh/Assignment-1  (Harinesh MA A).txt [hunk 23]
```diff
@@ -0,0 +1,139 @@
...
+23)Employee who joined latest
+Ans: select *
+    from emp
+    where hiredate = (select max(hiredate) from emp);
+
```
Comment (raviproengage) [inline Assignments/Harinesh/Assignment-1  (Harinesh MA A).txt:105]: ok

File: Assignments/Harinesh/Assignment-1  (Harinesh MA A).txt [hunk 24]
```diff
@@ -0,0 +1,139 @@
...
+24)Who is the oldest employee in the organization based on their hire date?
+Ans: select *
+    from emp
+    where hiredate = (select min(hiredate) from emp);
+
```
Comment (raviproengage) [inline Assignments/Harinesh/Assignment-1  (Harinesh MA A).txt:110]: ok

File: Assignments/Harinesh/Assignment-1  (Harinesh MA A).txt [hunk 25]
```diff
@@ -0,0 +1,139 @@
...
+25)Sort the employee based on their hire year (descending) and department (ascending) 
+Ans: select * 
+    from emp 
+    order by hiredate desc,deptno;
+
```
Comment (raviproengage) [inline Assignments/Harinesh/Assignment-1  (Harinesh MA A).txt:115]: not ok. sort on year and date.

select * from emp order by to_char(hiredate,'yyyy') desc, deptno;

File: Assignments/Harinesh/Assignment-1  (Harinesh MA A).txt [hunk 26]
```diff
@@ -0,0 +1,139 @@
...
+26. Employees who get salaries greater than or equal to the average salary of employees.
+Ans:select *
+    from emp
+    where sal >=(select avg(sal) from emp);
+
```
Comment (raviproengage) [inline Assignments/Harinesh/Assignment-1  (Harinesh MA A).txt:120]: ok

File: Assignments/Harinesh/Assignment-1  (Harinesh MA A).txt [hunk 27]
```diff
@@ -0,0 +1,139 @@
...
+27)Employees who get salary less than or equal to average salary of employees.
+Ans: select *
+    from emp
+    where sal <=(select avg(sal) from emp);
+
```
Comment (raviproengage) [inline Assignments/Harinesh/Assignment-1  (Harinesh MA A).txt:125]: ok

File: Assignments/Harinesh/Assignment-1  (Harinesh MA A).txt [hunk 28]
```diff
@@ -0,0 +1,139 @@
...
+28)Employees get salaries between 2000 and 4000.
+Ans: select *
+    from emp
+    where sal <=(select avg(sal) from emp);
+
```
Comment (raviproengage) [inline Assignments/Harinesh/Assignment-1  (Harinesh MA A).txt:130]: not ok.

select * from emp where sal between 2000 and 4000;

File: Assignments/Harinesh/Assignment-1  (Harinesh MA A).txt [hunk 29]
```diff
@@ -0,0 +1,139 @@
...
+29)Which employees get the highest and lowest salary?
+Ans: select * 
+    from emp 
+    where sal=(select max(sal) from emp) or sal=(select min(sal) from emp);
+
```
Comment (raviproengage) [inline Assignments/Harinesh/Assignment-1  (Harinesh MA A).txt:135]: ok

File: Assignments/Harinesh/Assignment-1  (Harinesh MA A).txt [hunk 30]
```diff
@@ -0,0 +1,139 @@
...
+30)We need to celebrate the joining month of the employees. We plan to buy a gift for each of them. How many gifts I need to buy for next month?
+Ans: select count(*) "no of gifts to buy" 
+    from emp 
+    where to_char(hiredate,'MON')='FEB';
```
Comment (raviproengage) [inline Assignments/Harinesh/Assignment-1  (Harinesh MA A).txt:139]: ok

**Review by raviproengage — COMMENTED** [review]
Good Submission. Keep it up!

File: Assignments/Harinesh/Assignment-1  (Harinesh MA A).txt (hunk 7 above)
Comment (Harinesh5) [reply Assignments/Harinesh/Assignment-1  (Harinesh MA A).txt:31]: ok sir

File: Assignments/Harinesh/Assignment-1  (Harinesh MA A).txt (hunk 8 above)
Comment (Harinesh5) [reply Assignments/Harinesh/Assignment-1  (Harinesh MA A).txt:35]: ok sir

File: Assignments/Harinesh/Assignment-1  (Harinesh MA A).txt (hunk 19 above)
Comment (Harinesh5) [reply Assignments/Harinesh/Assignment-1  (Harinesh MA A).txt:86]: ok sir

File: Assignments/Harinesh/Assignment-1  (Harinesh MA A).txt (hunk 25 above)
Comment (Harinesh5) [reply Assignments/Harinesh/Assignment-1  (Harinesh MA A).txt:115]: ok sir

File: Assignments/Harinesh/Assignment-1  (Harinesh MA A).txt (hunk 28 above)
Comment (Harinesh5) [reply Assignments/Harinesh/Assignment-1  (Harinesh MA A).txt:130]: I wrote the same query sir but made mistake while preparing document.

**Comment by Harinesh5** [comment]
Thank you sir

**Pull request closed without merging**